File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # loop thru nums list
        # keep track of index
        # subtract target - current num
        # if the complement is in the list
        # find index of complement
        # make sure complementDiff does not equal current num index
        # if equal, continue the loop
        # else return -- we've found two sums! 

        a = 0

        for i in range(len(nums)):
            complement = target - nums[i];

            if complement in nums:
                a = nums.index(complement)
                if a == i:
                    continue
                break
        return i, a 
            

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    // loop through nums array
    // create a hash map to keep track of previous visited nums = index

    // initialize complment = target - currentnum 
    // check if complement is in previous num
    // if yes, return value for corresponding key, and current index


    let previousNum = {};

    for (let i = 0; i < nums.length; i++) {
        let currentNum = nums[i];

        let complement = target - currentNum;

        if (complement in previousNum) {
            return [previousNum[complement], i];
        }

        previousNum[currentNum] = i; 
    }
};